wallet/src/Pos/Aeson/ClientTypes.hs

```diff
@@ -41,7 +41,31 @@ deriveJSON defaultOptions ''Wal
 deriveJSON defaultOptions ''Addr
 deriveJSON defaultOptions ''CHash
 deriveJSON defaultOptions ''CInitialized
-deriveJSON (customOptionsWithTag "groupingPolicy") ''InputSelectionPolicy
+
+-- NOTE(adinapoli): We need a manual instance to ensure we map @OptimizeForSize@
+-- to @OptimizeForHighThroughput@, for exchanges backward compatibility.
+instance ToJSON InputSelectionPolicy where
+  toJSON pol =
+      let renderPolicy = toJSON @Text $ case pol of
+              OptimizeForSecurity       -> "OptimizeForSize"
+              OptimizeForHighThroughput -> "OptimizeForHighThroughput"
+          in object [ "groupingPolicy" .= renderPolicy ]
+
+-- NOTE(adinapoli): Super lenient decoder to overcome the fact some revisions of
+-- the API were using a raw JSON String and later versions used on Object like
+-- `{"groupingPolicy": "blabla" }`
+instance FromJSON InputSelectionPolicy where
+    parseJSON (Object o) = fromRawPolicy =<< (o .: "groupingPolicy")
+    parseJSON (String rawPolicy) = fromRawPolicy rawPolicy
+    parseJSON x = typeMismatch "Not a valid InputSelectionPolicy" x
+
+fromRawPolicy :: Text -> Parser InputSelectionPolicy
+fromRawPolicy rawPolicy = case rawPolicy of
+    "OptimizeForSecurity"       -> pure OptimizeForSecurity
+    "OptimizeForHighThroughput" -> pure OptimizeForHighThroughput
+    "OptimiseForSize"           -> pure OptimizeForSecurity
```

Comment:
OptimizeForSize should map to OptimizeForHighThroughput, that's the one some exchanges are currently using.

Reply:
Clearly starting to get code fatigue @dcoutts . You are right all across the board.